File: iCinema.Infrastructure/Persistence/Models/Actor.cs
using System;
using System.Collections.Generic;

namespace iCinema.Infrastructure.Persistence.Models;

public partial class Actor
{
    public Guid Id { get; set; }

    public string FullName { get; set; } = null!;

    public string? Bio { get; set; }

    public string? PhotoUrl { get; set; }

    public virtual ICollection<MovieActor> MovieActors { get; set; } = new List<MovieActor>();
}

File: iCinema.UI/icinema_desktop/lib/features/reference/domain/actor.dart
class Actor {
  final String id;
  final String fullName;
  final String? bio;
  final String? photoUrl;

  Actor({required this.id, required this.fullName, this.bio, this.photoUrl});

  factory Actor.fromJson(Map<String, dynamic> json) => Actor(
        id: json['id'] as String,
        fullName: json['fullName'] as String,
        bio: json['bio'] as String?,
        photoUrl: json['photoUrl'] as String?,
      );
}
